### PR TITLE
build(release): add reno tools to build image and makefile

### PR DESCRIPTION
**Pull request by hairyhum**
## Change Overview

Tooling to make use of reno release notes generation tool

## Pull request type

Please check the type of change your PR introduces:
- [x] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

### CODE REVIEW

**Comment by hairyhum** [comment]
Added `RELEASE.md` with kanister release instructions. It currently describes the process with reno release notes, which are not merged yet.

File: .gitignore
Comment (viveksinghggits) [inline .gitignore:11]:
```suggestion
/releasenotes/config.yaml

```

File: Makefile
Comment (viveksinghggits) [inline Makefile:295]:
```suggestion
	@$(MAKE) run CMD="./build/reno_report.sh $(VERSION)"
	
```

File: releasenotes/README.md
Comment (viveksinghggits) [inline releasenotes/README.md:23]: whose responsibility will it be to create changelog file? when is it supposed to be created?

File: releasenotes/README.md
Comment (viveksinghggits) [inline releasenotes/README.md:13]: I think if we pass `--edit` to `reno new` it opens the newly created file so that we can edit it there only and don't have to open the newly created file again.
I think we should mention that and it would be helpful.

File: releasenotes/README.md
Comment (viveksinghggits) [inline releasenotes/README.md:16]:
```suggestion
Summary of the change should be added to this file to reflect the change and additional information such as deprecations or upgrade requirements.
```

?

File: reno.yaml
Comment (viveksinghggits) [inline reno.yaml:22]: `release_summary` should just be fillable by the maintainers right? Other contributors should just be able to add summary for other changes in other fields.

Comment (hairyhum) [reply reno.yaml:22]: Maybe. We can remove that from the template if you think it would be confusing.

File: releasenotes/README.md
Comment (hairyhum) [reply releasenotes/README.md:13]: Not sure that would work with docker container, but I can test it.

File: releasenotes/README.md
Comment (hairyhum) [reply releasenotes/README.md:23]: Changelog file would be created during the pre-release step automatically: https://github.com/kanisterio/kanister/pull/2774

But a person doing pre-release CAN do that manually as well.
I can mention that.

File: releasenotes/README.md
Comment (hairyhum) [reply releasenotes/README.md:13]: It is possible. Switched to that as this is most likely the intention of anyone running this command. Need to rebuild the build image though.

File: reno.yaml
Comment (hairyhum) [reply reno.yaml:22]: On the second thought, I don't think it's a good idea to change the template. Maintainers will most likely use the same command to create a note.

Comment (viveksinghggits) [reply reno.yaml:22]: hmm, My point is, release_summary is something that, if needed, should be part of entirely different PR (just the release_summary) and that PR would be raised once per release.
If every contributor gets that field, they might add things their, if we are ok to take that risk I am ok.

Comment (hairyhum) [reply reno.yaml:22]: We can remove it or ask to remove it at any point. There is a point during pre-release process when we're supposed to review the release notes.

File: CODE_REVIEW.md
Comment (viveksinghggits) [inline CODE_REVIEW.md:3]:
```suggestion
This document describes responsibilities of code reviewers when reviewing PRs.
```

File: CODE_REVIEW.md
Comment (viveksinghggits) [inline CODE_REVIEW.md:13]:
```suggestion
    - If description is missing but clear for reviewer, reviewer may request the author to add the description or edit description by themselves
```

File: CODE_REVIEW.md
Comment (viveksinghggits) [inline CODE_REVIEW.md:16]: It took me a second to realise what exactly we meant by `If PR changes documented behaviour`. Should we try to simplify it.
`If the PR changes a behaviour that is documented in our public docs`?
This should be applicable for something that is being  added (not updated)  and the documentation needs to be added for that, I think we should specify this as well.

File: CODE_REVIEW.md
Comment (viveksinghggits) [inline CODE_REVIEW.md:24]: Should we clarify that it doesn't mean that `Request changes` type of reviews should not be submitted. If someone thinks that some thing must be changed before the PR is merged, they should not hesitate adding `Request changes` type reviews.

File: CODE_REVIEW.md
Comment (viveksinghggits) [inline CODE_REVIEW.md:35]:
```suggestion
For PRs from Kanister developers, the author should add the `kueue` label after
```

File: CODE_REVIEW.md
Comment (viveksinghggits) [inline CODE_REVIEW.md:38]:
```suggestion
For PRs from community members, the reviewer should add the `kueue` label.
```

File: CODE_REVIEW.md
Comment (viveksinghggits) [inline CODE_REVIEW.md:60]: is punctuation required in the release notes?

File: build/reno_report.sh
Comment (viveksinghggits) [inline build/reno_report.sh:33]:
```suggestion
# Generate changelog for current version only:
```

File: build/reno_report.sh
Comment (viveksinghggits) [inline build/reno_report.sh:35]:
```suggestion
## Reno `--version` flag does not support "unreleased" setting and requires specific version, even if it's dynamic
```

File: CODE_REVIEW.md
Comment (hairyhum) [reply CODE_REVIEW.md:60]: We currently don't check grammar or punctuation. Do we need to? Can you give an example?

File: releasenotes/README.md
Comment (hairyhum) [reply releasenotes/README.md:13]: `reno new` now has `--edit` flag by default.

File: CODE_REVIEW.md
Comment (hairyhum) [reply CODE_REVIEW.md:24]: The issue with that is that person becomes a bottleneck for merging as mentioned below. That's why we should discourage it.

File: CODE_REVIEW.md
Comment (hairyhum) [reply CODE_REVIEW.md:61]: There is no requirement for grammar or punctuation in content at the moment.

Comment (viveksinghggits) [reply CODE_REVIEW.md:61]: cool, sounds good.

File: releasenotes/README.md
Comment (viveksinghggits) [inline releasenotes/README.md:6]:
```suggestion
Release notes are generated using [https://docs.openstack.org/reno/latest/](reno) and are stored in `releasenotes/notes` directory.

```

File: releasenotes/README.md
Comment (viveksinghggits) [inline releasenotes/README.md:19]:
```suggestion
When reviewing a PR, a reviewer should check if there are release notes added if necessary and either request or add a new note if they have push access to the branch.
```

File: releasenotes/README.md
Comment (viveksinghggits) [inline releasenotes/README.md:38]: should these sentences be together?
Also `CHANGELOG_CURRENT.md only containing changes for the current version.` doesn't seem right.

**Review by viveksinghggits — COMMENTED** [review]
LGTM, after addressing few minor comments.

File: releasenotes/README.md
Comment (hairyhum) [reply releasenotes/README.md:36]: > Also CHANGELOG_CURRENT.md only containing changes for the current version. doesn't seem right.

Can you elaborate?

**Review by e-sumin — APPROVED** [review]
LGTM

**Pull request merged**
